src/pip/_internal/req/constructors.py

```diff
@@ -236,26 +255,27 @@ def install_req_from_line(
         link = Link(name)
     else:
         p, extras_as_string = _strip_extras(path)
```

Comment:
Given what this PR is changing, it looks like everything in this `else` block after the above line should be made into a single function that returns what should be passed to `Link()` (or `None` if no `Link` object should be created). That function could in turn be broken up further (as you have already started to do with `_looks_like_path()`).

Reply:
Sorry for the delay. Is that what you had in mind?
https://github.com/pypa/pip/pull/6203/commits/7341528f5666f0f32a25c5d5fe00289d29097628

Reply:
Yes! Thank you. I do have some suggestions, but in the meantime I'd recommend adding at least some tests of both `_looks_like_path ()` and `_get_path_or_url()`. With these functions, it will be easier to test the various code paths (branches) because the functions are simpler.

Reply:
I'm making these tests right now :)